x/authz/keeper/keeper.go

```diff
@@ -135,14 +141,19 @@ func (k Keeper) DispatchActions(ctx sdk.Context, grantee sdk.AccAddress, msgs []
 		if err != nil {
 			return nil, sdkerrors.Wrapf(err, "failed to execute message; message %v", msg)
 		}
+
 		results[i] = msgResp.Data
 
 		// emit the events from the dispatched actions
 		events := msgResp.Events
 		sdkEvents := make([]sdk.Event, 0, len(events))
-		for i := 0; i < len(events); i++ {
-			sdkEvents = append(sdkEvents, sdk.Event(events[i]))
+		for _, event := range events {
+			e := event
+			e.Attributes = append(e.Attributes, abci.EventAttribute{Key: "authz_msg_index", Value: strconv.Itoa(i)})
```

Comment:
If we move back to normal event, we should possibly define this in an `events.go` as `AttributeKeyMsgIndex` or smth. Now we have this module combining TypedEvent and "Legacy" Events.

Reply:
agere. Im more inclined to revert the changes and then move to it later but need to make a team call on this.

Reply:
Huh? Where is there typed events used here?

Reply:
Not in your code, but here https://github.com/cosmos/cosmos-sdk/blob/main/x/authz/keeper/keeper.go#L195-L199 or there https://github.com/cosmos/cosmos-sdk/blob/main/x/authz/keeper/keeper.go#L221-L225

Reply:
ahh yeah, but we're just adding an attribute here though. What do you want to change?

Reply:
I was just noticing the usage of two types of events in the authz module :)